src/figuresClasses.ts

```diff
@@ -1,19 +1,96 @@
-export interface Figure {
+export type Shape = 'triangle'|'circle'|'rectangle';
+export type Color = 'red'|'green'|'blue';
 
+export interface Figure {
+  shape: string;
+  color:string;
+  getArea():number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  color: Color = 'red';
+
+  a: number;
+
+  b: number;
+
+  c:number;
 
-export class Circle {
+  constructor(color: Color, a:number, b:number, c:number) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
 
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('ERROR, side length can not be 0 or negative');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('ERROR, it is not triangle');
+    }
+  }
```

Comment:
I think you can combine this into one if with one Error
But it is not necessary

Reply:
I put two different errors because we have two checks